website/docs/about_code_generation.mdx

```diff
@@ -59,9 +59,42 @@ as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pu
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## Provider type for code generation
+
+### \[Sync/Async]\[Stateless/Stateful]Provider
+
+- <i><u>Note:</u></i> This is just a way of thinking, not actual types. You don’t need to pick
+  a type when using code generation variant anyway!
+
+- `Sync/Async` is being decided by your return type, either of your function or the [build] method
+  in case of using a class (stateful):
+   - `Sync` ⇒ T `Async` ⇒ Future<T\> or Stream<T\>
+
+- `Stateless/Stateful` is being decided by whether you need to expose methods to modify the state or not (side-effects):
+   - `Stateless` ⇒ use a function. `Stateful` ⇒ use a class with [build] method + side effect methods.
+
+
+### Migrate from non code generation variant:
+When using non code generation variant, it is necessary to manually determine the type of your provider.
+The following are the corresponding options for transitioning into code generation variant:
+
+- Sync Stateless Provider ⇒ [Provider]
```

Comment:
Rather than names which likely won't mean anything for most people, I think code snippets would be better.